### ISSUE TITLE
Decompile clz in a clearer manner

### ISSUE THREAD

**Issue by copumpkin** [NationalSecurityAgency/ghidra#6003]
**Is your feature request related to a problem? Please describe.**

I'm analyzing some ARM code that uses a lot of `clz` in it. I didn't initially notice the `clz` in the disassembly and spent quite some time trying to figure out what the obscure chain of bit manipulations was doing before deciding to see what the disassembly looked like and noticing it was a technically correct but very confusing attempt to talk about a primitive not available directly in C.

For context, a single `clz` instruction will expand to a decompilation that looks like this:

```
  uVar43 = uVar43 | uVar43 >> 2;
  uVar43 = uVar43 | uVar43 >> 4;
  uVar43 = uVar43 | uVar43 >> 8;
  uVar43 = ((uVar43 & 0xaaaaaaaaaaaaaaaa) >> 1) + (uVar43 & 0x5555555555555555);
  uVar43 = ((uVar43 & 0xcccccccccccccccc) >> 2) + (uVar43 & 0x3333333333333333);
  lVar42 = ((uVar43 & 0xf0f0f0f0f0f0f0f0) >> 4) + (uVar43 & 0xf0f0f0f0f0f0f0f);
```

**Describe the solution you'd like**

Introduce some sort of primitive way of talking about `clz` and other such operations. It's more useful to me as a reader to know that I'm counting leading zeroes than to have to reverse engineer obscure bitwise arithmetic that wasn't in the original program.

**Describe alternatives you've considered**

This could be an optional feature if for whatever reason people preferred the current decompilation. Or wrap the generated code in a pair of comments clarifying what the big lump of code is, ideally making it clear what the operand to `clz` was, because it takes some staring to figure it out in the current decompilation.

**Additional context**

I found https://github.com/NationalSecurityAgency/ghidra/issues/2121 which seemed to be talking about something similar. I'm not familiar enough with Ghidra's implementation to know how it relates though.

**Comment by LukeSerne** [comment]
In Ghidra 10.3, the `lzcount` pcode operation was introduced. It seems to me that this pcode operation is also used when describing the semantics of the `clz` ARM instruction. As such, you should see `lzcount` show up in the decompiler output of the latest Ghidra versions.

It would be helpful if you could share the Ghidra version you're using and the binary that you're analysing where you're seeing this. If you cannot share the full binary, the "Debug Function Decompilation" xml of the function containing the `clz` operation will suffice. You can create this xml file by going to the function containing the `clz` instruction, clicking the downwards pointing triangle in the upper right corner of the decompiler window and clicking the "Debug Function Decompilation" menu item.

**Comment by mumbel** [comment]
#5790

**Comment by GhidorahRex** [comment]
As @mumbel noted, this should be fixed in 10.4.1. I'd suggest trying the updated ARM module and it should be resolved.

**Comment by copumpkin** [comment]
Nice. Is there an easy way to try out the new module without building Ghidra from source?

**Comment by mumbel** [comment]
SLEIGH is compiled by ghidra at startup either from updated timestamps or not yet compiled module producing sla files.  You can apply the AARCH64 changes from that fixed by commit and completely restart ghidra and that should be it

You can also compile SLEIGH manually with `<ghidra_install>/support/sleigh <path to slaspec>` (with various debug flags) if it does not open the project file after reloading.

**Comment by copumpkin** [comment]
Thanks, that worked great!

**Comment by LukeSerne** [comment]
Also note that the decompilation of `ubfiz` not being "helpful for understanding" will be fixed if/when #5772 is merged.

**Comment by copumpkin** [comment]
Awesome, I'll watch for that one too! Thanks again!
